### PR TITLE
Enable metrics collection of and introduce dashboard for envoy proxy side car used in reversed vpn

### PR DESCRIPTION
**Pull request by ScheererJ**
Co-authored-by: Sebastian Stauch <sebastian.stauch@sap.com>

**How to categorize this PR?**

/area networking
/area monitoring
/kind enhancement

**What this PR does / why we need it**:
Prior to this change there is very limited visibility into what is happening on the envoy proxy side car of the `vpn-seed-server` in the shoots' control planes. This change enables metric collection from the envoy proxy side car as well as exposes the metrics in form of a grafana dashboard in the operator grafana of each shoot.
With the metrics and the dashboard it is hopefully easier going forward to understand some problems of the reversed vpn.

**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:

**Release note**:

```other operator
Enable metrics collection of and introduce dashboard for envoy proxy side car used in reversed vpn
```

### CODE REVIEW

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml [hunk 1]
```diff
@@ -627,6 +627,28 @@ data:
         replacement: 'shoot_node_logging_incoming_$1'
 {{- end }}
 
+{{- if .Values.reversedVPN.enabled }}
+    - job_name: reversed-vpn-envoy-side-car
+      honor_labels: false
```
Comment (istvanballok) [inline charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml]: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config default is false
```suggestion
```

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml [hunk 2]
```diff
@@ -627,6 +627,28 @@ data:
...
+        - __meta_kubernetes_service_name
+        - __meta_kubernetes_service_port_name
+        action: keep
+        regex: vpn-seed-server;metrics
+      metric_relabel_configs:
+      - source_labels: [__name__]
+        target_label: __name__
+        regex: '(.*)'
+        action: replace
+        replacement: 'reversed_vpn_$1'
```
Comment (istvanballok) [inline charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml]:
```suggestion
```
In prometheus it is idiomatic to keep the original metric name and just use labels for further differentiation. You could use the default `job` label to filter for the envoy metrics of your own deployment.

File: charts/seed-monitoring/charts/core/charts/prometheus/values.yaml [hunk 3]
```diff
@@ -260,6 +263,56 @@ allowedMetrics:
   - envoy_tcp_downstream_cx_rx_bytes_total
   - envoy_tcp_downstream_cx_total
   - envoy_tcp_downstream_cx_tx_bytes_total
+  reversedVPN:
+  - reversed_vpn_envoy_cluster_external_upstream_rq
```
Comment (istvanballok) [inline charts/seed-monitoring/charts/core/charts/prometheus/values.yaml]: the prefix has to be removed here as well

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml (hunk 2 above)
Comment (ScheererJ) [reply charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml]: Adapted accordingly with Sebastian's latest commit.

File: charts/seed-monitoring/charts/core/charts/prometheus/values.yaml (hunk 3 above)
Comment (ScheererJ) [reply charts/seed-monitoring/charts/core/charts/prometheus/values.yaml]: Adapted accordingly with Sebastian's latest commit.

File: charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml (hunk 1 above)
Comment (ScheererJ) [reply charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml]: Removed unnecessary line with latest commit.

**Review by istvanballok — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**